ludwig/schema/utils.py

```diff
@@ -242,7 +242,7 @@ def ReductionOptions(default: Union[None, str] = None, description="", parameter
 @DeveloperAPI
 def RegularizerOptions(
     default: Union[None, str] = None,
-    allow_none: bool = True,
+    allow_none: bool = False,
```

Comment:
Having default of `None` is.contradictory. If we remove `None`, we should choose a different default.